src/ddoc/highlight.d

```diff
@@ -133,6 +133,28 @@ $(D_CODE $(D_KEYWORD unittest) {
 	assert(r1 == e1, r1);
 }
 
+unittest {
+	auto s = `
+        ---------
+        asm pure nothrow @nogc @trusted
+        {
+            // the compiler does not check the attributes
+            ret;
+        }
+        ---------
+`;
+	auto e = `
+        $(D_CODE $(D_KEYWORD asm) $(D_KEYWORD pure) $(D_KEYWORD nothrow) @nogc @trusted
```

Comment:
The test you refered to earlier should be:

````
/**
 * [...]
 *     ----
 *     int a;
 *     int b;
 *     ----
 * [...]
 */
````

In the above example you want the example itself to be indented, but not the content.
Removing **comments** indentation is not the purpose of this change (but thanks for bringing that up). I only aim to remove extra unittest indent here, the change for comments is a bit more involved (like removing leading '*' and '+') and should happen before this phase.